ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -140,9 +320,21 @@ public int compareTo(StoreKey o) {
 
     int result = version.compareTo(other.version);
     if (result == 0) {
```

Comment:
These extra information that we are adding (account id, flags etc.) do not necessarily have to get accounted in the comparison. This method could actually go untouched as you will not have blob ids that just differ in their flags or account ids.

This is probably okay - but do make sure that the comparison stays exactly the same for V1 blob ids (it seems to), otherwise the lookups in existing segments will go wrong.